### PR TITLE
adding VBFNLO and Hjets++ to the externals linked to Herwig7

### PR DESCRIPTION
**Pull request by Andrej-CMS**

### CODE REVIEW

File: vbfnlo.spec [hunk 1]
```diff
@@ -0,0 +1,24 @@
+### RPM external vbfnlo 3.0.0beta5
+Source: http://www.itp.kit.edu/~vbfnloweb/archive/vbfnlo-%{realversion}.tgz
```
Comment (smuzaffar) [inline vbfnlo.spec]: @Andrej-CMS , why not use https://github.com/vbfnlo/vbfnlo  to download it? You can use
```
git+https://github.com/vbfnlo/%{n}.git?obj=master/v%{realversion}&export=%{n}-%{realversion}&output=/%{n}-%{realversion}.tgz
```

and then can do
```
%setup -q -n %{n}-%{realversion}
```

**Comment by smuzaffar** [comment]
@Andrej-CMS , do we have any tests in CMSSW to test hjets?

**Comment by Andrej-CMS** [comment]
@smuzaffar 
thanks for your comments!
I changed to download from git. We don't have the hjets test yet. Do you want me to add a runTheMatrix command?

**Comment by smuzaffar** [comment]
@Andrej-CMS , at least a unit test to make sure that our distribution of hjets is good/working

**Comment by Andrej-CMS** [comment]
@smuzaffar 
sure, I can add that. Could you maybe in the meantime try to build this here?
I had quite some issues building these packages, but this version here at least worked for me locally

**Comment by smuzaffar** [comment]
please test

File: vbfnlo.spec (hunk 1 above)
Comment (Andrej-CMS) [reply vbfnlo.spec]: Dear @smuzaffar ,
I get now an error that states that the Manual.pdf file cannot be compiled from TeX file. Do we have in cmsdist some latex package that need to build first?
[vbfnlobuild.txt](https://github.com/cms-sw/cmsdist/files/2940692/vbfnlobuild.txt)

Comment (Andrej-CMS) [reply vbfnlo.spec]: @smuzaffar 
do you have any idea in regards of the question above?

**Comment by mrodozov** [comment]
please test

File: vbfnlo.spec [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+BuildRequires: autotools
+
+%prep
+%setup -q -n %{n}-%{realversion}
+
+%build
+CXX="$(which g++) -std=c++11"
+CC="$(which gcc)"
+FC="$(which gfortran)"
+
```
Comment (smuzaffar) [inline vbfnlo.spec]: @Andrej-CMS , as mentioned in https://github.com/vbfnlo/vbfnlo/blob/master/README.md , you need to run `autoreconf -vi` before running `configure`

File: vbfnlo.spec (hunk 1 above)
Comment (smuzaffar) [reply vbfnlo.spec]: @Andrej-CMS , I do not see build errors if I run `autoreconf -vi` before configure

**Comment by smuzaffar** [comment]
please test
ok I have updated the PR to run `autoreconf -vi`

**Comment by smuzaffar** [comment]
assign generators

**Comment by smuzaffar** [comment]
@gudrutis , @mrodozov can you please check if this works for aarch64 and gcc820 Ibs

**Comment by Andrej-CMS** [comment]
@smuzaffar 
thanks a lot for checking. That's strange that I got build error when I added autoreconf -vi. Do you build on LXPLUS or do you have dedicated build machines for that?

**Comment by mrodozov** [comment]
the pr builds on arm64 and with gcc8. from building view point it's fine

**Comment by smuzaffar** [comment]
@Andrej-CMS , we build on our build machines but it should work on lxplus too

**Comment by Andrej-CMS** [comment]
@smuzaffar 
In the past, it worked quite well to build on lxplus, but these days I always get some errors. With the version  of the externals from the environment below, I get the error in the file below (libHelas cannot be built).
SCRAM_ARCH=slc6_amd64_gcc700;PKGTOOLS_TAG=V00-32-XX;CMSDIST_TAG=IB/CMSSW_10_5_X/gcc700;RELEASE_BRANCH=CMSSW_10_5_X;RELEASE_QUEUE=CMSSW_10_5_X;DOCKER_IMG=cmssw/slc6:latest;PR_TESTS=1;

Are there any good and stable machines that a normal user can use to build the externals efficiently? I keep spending a lot of time trying to build on LXPLUS or our local cluster and get errors, I cannot reproduce here (probably because of some non-matching versions of software installed)

[buildCMSSW_lxplus.txt](https://github.com/cms-sw/cmsdist/files/2967371/buildCMSSW_lxplus.txt)

**Comment by smuzaffar** [comment]
@Andrej-CMS , you can use cmsdev20 to cmsdev25. These are CentOS7 machines, so build for slc7_amd64_gcc700 arch.

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
+externals
@alberto-sanchez,@perrozzi,@efeyazgan,@qliphy can you please sign this?

**Comment by qliphy** [comment]
+1

**Comment by smuzaffar** [comment]
@fabiocos , this is ready to go in too.

**Comment by fabiocos** [comment]
+1

**Comment by smuzaffar** [comment]
@Andrej-CMS , HJets failed to compile with GCC 9

https://cmssdt.cern.ch/SDT/jenkins-artifacts/build-any-ib/CMSSW_11_0_X_2019-06-05-2300/slc7_amd64_gcc900/76593/external/hjets/1.2/log
```
  CXX      HJetsProcessInfo.lo
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/Config/Complex.h:16,
                 from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/Config/ThePEG.h:31,
                 from HJetsProcessInfo.h:13,
                 from HJetsProcessInfo.cc:11:
/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/gcc/9.0.0-nmpfii/include/c++/9.1.1/complex: In instantiation of 'std::complex<double>& std::complex<double>::operator=(const std::complex<_Tp>&) [with _Tp = ThePEG::Qty<0, 0, 0>]':
HJetsProcessInfo.cc:240:93:   required from here
/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/gcc/9.0.0-nmpfii/include/c++/9.1.1/complex:1319:19: error: 'const struct std::complex<ThePEG::Qty<0, 0, 0> >' has no member named '__rep'
 1319 |    _M_value = __z.__rep();
      |               ~~~~^~~~~
make[1]: *** [HJetsProcessInfo.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/Config/Complex.h:16,
                 from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/Config/ThePEG.h:31,
                 from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/Interface/Interfaced.h:13,
                 from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/Handlers/HandlerBase.h:13,
                 from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/thepeg/2.1.5/include/ThePEG/MatrixElement/Amplitude.h:14,
                 from /build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/herwigpp/7.1.5/include/Herwig/MatrixElement/Matchbox/Base/MatchboxAmplitude.h:15,
                 from AmplitudeBase.h:16,
                 from AmplitudeBase.cc:15:
/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/gcc/9.0.0-nmpfii/include/c++/9.1.1/complex: In instantiation of 'std::complex<double>& std::complex<double>::operator=(const std::complex<_Tp>&) [with _Tp = ThePEG::Qty<0, 0, 0>]':
AmplitudeBase.cc:587:70:   required from here
/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/external/gcc/9.0.0-nmpfii/include/c++/9.1.1/complex:1319:19: error: 'const struct std::complex<ThePEG::Qty<0, 0, 0> >' has no member named '__rep'
 1319 |    _M_value = __z.__rep();
      |               ~~~~^~~~~
make[1]: *** [AmplitudeBase.lo] Error 1
```

**Comment by smuzaffar** [comment]
@fabiocos , @Andrej-CMS  I am afraid we need to revert it back as it fails to build for GCC 9

**Comment by fabiocos** [comment]
@Andrej-CMS I think that the code developers/maintainers should be notified about this issue, adapting also the extensions to the previous fix.

**Pull request merged**